xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -0,0 +1,246 @@
+            pfp: "sub_logo",
+            bio: "Wow, it's your own profile! With its own codepath!",
+            category: .you
+        )
+        
+        return UserProfileContentPayload(
+            profile: profile,
+            statistics: UserProfileStatistics(
+                noteCount: entries.count,
+                backlinkCount: -1, // TODO: populate with real count
+                followingCount: following.count
+            ),
+            following: following,
+            entries: entries,
+            isFollowingUser: false
+        )
+    }
+    
+    nonisolated func getOwnProfilePublisher() -> AnyPublisher<UserProfileContentPayload, Error> {
+        Future.detached {
+            return try await self.getOwnProfile()
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    /// Retrieve all the content for the passed user's profile view, fetching their profile, notes and address book.
+    func getUserProfile(petname: Petname) async throws -> UserProfileContentPayload {
+        let sphere = try await self.noosphere.traverse(petname: petname)
+        let identity = try await sphere.identity()
+        let localAddressBook = AddressBook(sphere: sphere)
+        let following: [StoryUser] = try await self.getFollowingList(
+            sphere: sphere,
+            localAddressBook: localAddressBook,
+            traversalPath: petname
+        )
+        
+        // Detect your own profile and intercept
+        guard try await self.noosphere.identity() != identity else {
+            return try await getOwnProfile()
+        }
+        
+        guard let did = Did(identity) else {
+            throw UserProfileServiceError.invalidSphereIdentity
+        }
+        
+        let notes = try await sphere.list()
+        let isFollowing = await self.addressBook.isFollowingUser(did: did)
+        
+        var entries: [EntryStub] = []
+        
+        for slug in notes {
+            let slashlink = Slashlink(slug: slug)
+            let memo = try await sphere.read(slashlink: slashlink)
+            
+            guard let memo = memo.toMemo() else {
+                continue
+            }
+
+            entries.append(
+                EntryStub(
+                    address: Slashlink(petname: petname, slug: slug).toPublicMemoAddress(),
+                    excerpt: memo.excerpt(),
+                    modified: memo.modified
+                )
+            )
+        }
+        
+        let profile = UserProfile(
+            did: did,
+            petname: petname,
+            // TODO: replace with _profile_.json data
+            pfp: "pfp-dog",
+            bio: "Pretend this comes from _profile_.json",
+            category: .human
+        )
+        
+        return UserProfileContentPayload(
+            profile: profile,
+            statistics: UserProfileStatistics(
+                noteCount: entries.count,
+                backlinkCount: -1, // TODO: populate with real count
```

Comment:
I don't know how to populate this right now so leaving as -1 since it's offensively wrong looking in the UI.